tests/playbooks/tests_route_next_hop_ethernet.yml

```diff
@@ -0,0 +1,188 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Test configuring ethernet devices
+  hosts: all
+  vars:
+    type: veth
+    interface0: ethtest0
+    interface1: ethtest1
+
+  tasks:
+    - name: Set type and interface0
+      set_fact:
+        type: "{{ type }}"
+        interface: "{{ interface0 }}"
+    - name: Show interfaces
+      include_tasks: tasks/show_interfaces.yml
+    - name: Manage test interface
+      include_tasks: tasks/manage_test_interface.yml
+      vars:
+        state: present
+    - name: Assert device is present
+      include_tasks: tasks/assert_device_present.yml
+    - name: Set type and interface1
+      set_fact:
+        type: "{{ type }}"
+        interface: "{{ interface1 }}"
+    - name: Show interfaces
+      include_tasks: tasks/show_interfaces.yml
+    - name: Manage test interface
+      include_tasks: tasks/manage_test_interface.yml
+      vars:
+        state: present
+    - name: Assert device is present
+      include_tasks: tasks/assert_device_present.yml
+    - name: Run the test
+      block:
+        - name: Configure the IP addresses
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface0 }}"
+                interface_name: "{{ interface0 }}"
+                state: up
+                type: ethernet
+                autoconnect: true
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.3/24
+                    - 2001:db8::2/32
+                  route:
+                    - network: 198.51.10.64
+                      prefix: 26
+                      gateway: 198.51.100.6
+                      metric: 4
+                    - network: 2001:db6::4
+                      prefix: 128
+                      gateway: 2001:db8::1
+                      metric: 2
+              - name: "{{ interface1 }}"
+                interface_name: "{{ interface1 }}"
+                state: up
+                type: ethernet
+                autoconnect: true
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.6/24
+                    - 2001:db8::4/32
+                  route:
+                    - network: 198.51.12.128
+                      prefix: 26
+                      gateway: 198.51.100.1
+                      metric: 2
+        - name: Get the IPv4 routes from the route table main
+          command: ip -4 route
+          register: route_table_main_ipv4
+          ignore_errors: true
+          changed_when: false
+
+        - name: Get the IPv6 routes from the route table main
+          command: ip -6 route
+          register: route_table_main_ipv6
+          ignore_errors: true
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv4
```

Comment:
IMHO it is cleaner to keep getting and assertions together, so first get IPv4 then assert IPv4, then the same for IPv6. But this is a nitpick.